integration-tests/tests/serial_number_test.rs

```diff
@@ -0,0 +1,99 @@
+mod common;
+use anyhow::{bail, Context, Result};
+use common::{Binary, LogSide, TestContext};
+use lazy_static::lazy_static;
+use regex::Regex;
+use std::{io::Read, time::Duration};
+
+const L: LogSide = LogSide::Test;
+
+#[tokio::test]
+async fn test_device_info_serialnumber_ext() -> Result<()> {
+    let mut ctx = TestContext::new().context("Error building test context")?;
+
+    //Importtant to add cfg serial_number_format as SerialNumberExt to test this feature
+    let mfg_server = ctx
+        .start_test_server(
+            Binary::ManufacturingServer,
+            |cfg| {
+                cfg.prepare_config_file(None, |cfg| {
+                    cfg.insert("rendezvous_port", "1337");
+                    cfg.insert("diun_key_type", "FileSystem");
+                    cfg.insert("serial_number_format", "SerialNumberExt");
+                    Ok(())
+                })
+            },
+            |_| Ok(()),
+        )
+        .context("Error creating manufacturing server")?;
+    ctx.wait_until_servers_ready()
+        .await
+        .context("Error waiting for servers to start")?;
+
+    let diun_cert_path = ctx
+        .keys_path()
+        .join("diun_cert.pem")
+        .to_string_lossy()
+        .to_string();
+
+    let client_result = ctx
+        .run_client(
+            Binary::ManufacturingClient,
+            Some(&mfg_server),
+            |cfg| {
+                cfg.env("DEVICE_CREDENTIAL_FILENAME", "devicecredential.dc")
+                    .env("DIUN_PUB_KEY_INSECURE", "true")
+                    .env("DIUN_PUB_KEY_ROOTCERTS", diun_cert_path);
+                Ok(())
+            },
+            Duration::from_secs(5),
+        )
+        .context("Error running manufacturing client")?;
+    client_result
+        .expect_success()
+        .context("Manufacturing client failed")?;
+
+    let dc_path = client_result.client_path().join("devicecredential.dc");
+    L.l(format!("Device Credential should be in {:?}", dc_path));
+
+    let owner_output = ctx
+        .run_owner_tool(
+            client_result.client_path(),
+            &["dump-device-credential", dc_path.to_str().unwrap()],
+        )
+        .context("Error running dump-device-credential")?;
+    owner_output
+        .expect_success()
+        .context("Dump-device-credential failed")?;
+
+    for line in owner_output.raw_stdout().split(|&b| b == b'\n') {
```

Comment:
if the `owner_output` is empty for some reason the test will exit with the `Ok(())`, we should fail in that case

Reply:
taken care here